spec/queries/filtered_notes_query_spec.rb

```diff
@@ -0,0 +1,26 @@
+require "rails_helper"
+
+describe FilteredNotesQuery do
+  subject(:query) { described_class.new(Note, params) }
+
+  def note(title, content)
```

Comment:
What do you think about using `let`?
```ruby
let!(:first_note) { create :note, title: "One", content: "Two" }
let!(:second_note) { create :note, title: "Two", content: "One" }
let!(:third_note) { create :note, title: "Three", content: "Three" }

let(:filter_params) { {} }

describe ".all" do
  subject { query.all }

  it { is_expected.to match_array([first_note, second_note, third_note]) }

  context "with query filter parameter" do
    let(:filter_params) { { query: "one" } }

    it { is_expected.to match_array([first_note, second_note]) }
  end
end
```

Reply:
I'm not sure that the test will work correctly, since if we don't access some of those, then it won't be created and the query won't filter anything. Am I right? Also I can build them with `let` and persist in `before` block.